### ISSUE TITLE
Extend the REST API to allow queries for packages added since a given date

### ISSUE THREAD

**Issue by warkentyne** [dart-lang/pub-dev#7539]
There are currently over 50,000 results when you get the package names using `https://pub.dev/api/package-names`. It would be nice if the endpoint allowed the addition of a query string parameter to request only packages that have versions published since a given date. For example: `https://pub.dev/api/package-names?after=<UTC date string>`

By the way, `nextUrl` always seems to be null.

**Comment by isoos** [comment]
I don't think we have plans for adding such parameter (it requires different index/caching), but if/when we are changing the backing data structure, we may add it.

> By the way, nextUrl always seems to be null.

As noted on the page: https://pub.dev/help/api "The order of the packages, the number of packages on a response page, or the URL structure of "`nextUr`l" may change without notice."

In other words: we may introduce paginated package names at any time in the future (esp. as the number of names grows), so you should not ignore the `nextUrl` parameter.
